### PR TITLE
Add distillation tutorial

### PR DESCRIPTION
**Pull request by baiyfbupt**
Preview link: https://baiyfbupt.github.io/PaddleSlim/tutorials/distillation_demo/

### CODE REVIEW

File: docs/docs/tutorials/distillation_demo.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+本示例将介绍如何使用PaddleSlim蒸馏接口来对模型进行蒸馏训练
```
Comment (wanghaoshuang) [inline docs/docs/tutorials/distillation_demo.md]: 请加个句号？

File: docs/docs/tutorials/distillation_demo.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    # student model definition
+    model = MobileNet()
+    out = model.net(input=image, class_dim=1000)
+    cost = fluid.layers.cross_entropy(input=out, label=label)
+    avg_cost = fluid.layers.mean(x=cost)
+```
+
+### 2. 定义teacher_program
+
+在定义好teacher_program后，可以一并加载训练好的pretrained_model
```
Comment (wanghaoshuang) [inline docs/docs/tutorials/distillation_demo.md]: teacher_program最好用引起来把？`teacher_program`

Comment (wanghaoshuang) [reply docs/docs/tutorials/distillation_demo.md]: 注意加句号。下同。

File: docs/docs/tutorials/distillation_demo.md (hunk 1 above)
Comment (baiyfbupt) [reply docs/docs/tutorials/distillation_demo.md]: fixed, thanks

File: docs/docs/tutorials/distillation_demo.md (hunk 2 above)
Comment (baiyfbupt) [reply docs/docs/tutorials/distillation_demo.md]: fixed

**Review by wanghaoshuang — APPROVED** [review]
LGTM.

**Pull request merged**